### PR TITLE
Provide automatic removal of pkg-resources==0.0.0

### PR DESCRIPTION
**Pull request by nvtkaszpir**
Should help to deal with that pretty paintuly issue under Ubuntu/Debian
family:
https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1635463

### CODE REVIEW

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 1]
```diff
@@ -69,6 +69,11 @@ def fix_requirements(f):
     else:
         rest = []
 
+    for requirement in requirements:
+        if b'pkg-resources' in requirement.name:
+            if b'0.0.0' in requirement.value:
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py]: I think this makes more sense?

```python
if requirement.value == b'pkg-resources==0.0.0\n':
    ...
```

**Comment by asottile** [comment]
Otherwise looks good, shame debian botched the metadata here :(

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 2]
```diff
@@ -69,10 +69,11 @@ def fix_requirements(f):
...
         rest = []
 
+    # find and remove pkg-resources==0.0.0
+    # which is automatically added by broken pip package under Debian
     for requirement in requirements:
-        if b'pkg-resources' in requirement.name:
-            if b'0.0.0' in requirement.value:
-                requirements.remove(requirement)
+        if requirement.value == b'pkg-resources==0.0.0\n':
+            requirements.remove(requirement)
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py]: actually, now that I look at it I think this loop results in UB (modifying list while iterating over it) -- you can avoid this with:

```python
requirements = [
    requirement for requirement in requirements
    if requirement.value != b'pkg-resources==0.0.0\n'
]
```

**Comment by asottile** [comment]
Thanks for the feature!

**Comment by nvtkaszpir** [comment]
Thanks for the comments. Looking at it now all those comments and CI failures ... it's a sign I should go to sleep ;)

**Comment by asottile** [comment]
This has been (finally!) released as part of [v1.3.0](https://github.com/pre-commit/pre-commit-hooks/releases/tag/v1.3.0)! 🎉

**Pull request merged**